lib/SIL/SILVerifier.cpp

```diff
@@ -2311,9 +2311,18 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
         if (inst->isTypeDependentOperand(*use))
           continue;
         switch (inst->getKind()) {
+        case SILInstructionKind::MarkDependenceInst:
+          break;
         case SILInstructionKind::ApplyInst:
         case SILInstructionKind::TryApplyInst:
         case SILInstructionKind::PartialApplyInst:
+          // Non-Mutating set pattern that allows a inout (that can't really
+          // write back.
+          if (auto *AI = dyn_cast<ApplyInst>(inst)) {
+            if (isa<PointerToThinFunctionInst>(AI->getCallee())) {
```

Comment:
I don't get it. What does the function representation have to do with consuming-or-mutating arguments? I think this warrants a little more explanation.